tests/test_quote.py

```diff
@@ -373,8 +348,24 @@ def test_add_quotes(
     m = parse_mention(mention, database)
     add_quote(requester, m, quote, time, database)
     q = quotes_query(new_id, database).one_or_none()
+    actual = quote_str(q)
+
+    assert actual == expected
+    assert database.query(Quote).count() == db_size
+
+
+@pytest.mark.parametrize(
+    ["requester", "mention", "quote", "time", "error", "db_size"],
+    ADD_FAIL.values(),
+    ids=ADD_FAIL.keys(),
+)
+def test_add_fails(database, requester, mention, quote, time, error, db_size):
+    m = parse_mention(mention, database)
+    try:
+        add_quote(requester, m, quote, time, database)
+    except QuoteException as e:
+        assert e.err == error
```

Comment:
The [pytest.raises context manager](https://docs.pytest.org/en/6.2.x/reference.html#pytest.raises) would be a better fit here. You can then perform an assertion on the exception to check its value.